src/components/instance-form-builder/GeneralInfoForm.jsx

```diff
@@ -199,6 +199,25 @@ class GeneralInfoForm extends React.Component {
             }
         };
 
+        const testConnection = async () => {
```

Comment:
Move it to a class method to lighten the render method and to keep the prop constant.

Reply:
This was my first approach but I lacked the `fields` to perform the form validation before checking the connection (to make sure url, username, pass, were entered). I could move the bulk of the method to a class method and just call the form validation and then the class method from the render